src/actions/actions.ts

```diff
@@ -962,6 +962,16 @@ class CommandEscInsertMode extends BaseCommand {
 
     vimState.currentMode = ModeName.Normal;
 
+    // If we wanted to repeat this insert, now is the time to do it. Insert
+    // count amount of these strings before returning back to normal mode
+    if (vimState.recordedState.count > 1) {
+      for (let i = 0; i < vimState.recordedState.count - 1; i++) {
+        vimState.recordedState.transformations.push({
```

Comment:
IMHO `vimState.historyTracker.currentContentChanges` is what you need. Each time we enter Insert Mode, `vimState.historyTracker.currentContentChanges` will be refreshed so here at this line, you can try to duplicate this array and ask the transformations to apply the content changes for you, similar to 

```
    vimState.recordedState.transformations.push({
      type: "macro",
      register: register,
      replay: "contentChange"
    });
```

But I'm afraid we need a new transformation type as right now I don't think we can replay contentChange directly